rust/sbp2json/tests/test_error.rs

```diff
@@ -0,0 +1,27 @@
+mod common;
+
+use crate::common::{data_path, find_project_root};
+use assert_cmd::cargo::CommandCargoExt;
+use assert_cmd::Command;
+
+/// Asserts termination of json decoder on broken inputs
+#[test]
+fn test_json_converters_termination() {
```

Comment:
added a couple of simple tests for the termination case that wasn't previously handled and some errors
think we won't need much of the error message tests since `serde_json` should cover them

@silverjam